src/tests/bptree.test.ts

```diff
@@ -1,49 +1,102 @@
-    this._root = initialRoot;
-  }
+	beforeEach(() => {
+		mockDataFileResolver = async ({ start, end }) => {
+			return {
+				data: new ArrayBuffer(0),
+				totalLength: 0,
+			};
+		};
+	});
 
-  setRoot(mp: MemoryPointer): void {
-    this._root = mp;
-  }
+	it("should read a bptree and find items", async () => {
+		mockRangeResolver = async ({ start, end }) => {
+			const indexFile = await readBinaryFile("bptree_1.bin");
+			const slicedPart = indexFile.slice(start, end + 1);
 
-  root(): MemoryPointer {
-    return this._root;
-  }
-}
+			const arrayBuffer = slicedPart.buffer.slice(
+				slicedPart.byteOffset,
+				slicedPart.byteOffset + slicedPart.byteLength
+			);
+
+			return {
+				data: arrayBuffer,
+				totalLength: arrayBuffer.byteLength,
+			};
+		};
+
+		const page = new testMetaPage({ offset: 8192n, length: 88 });
+		const bptree = new BPTree(mockRangeResolver, page, mockDataFileResolver);
 
-describe("test compare bytes", () => {
-  let buffer: Uint8Array;
-  let resolver: RangeResolver;
+		let idx = 1;
+		for (const value of ["hello", "world", "moooo", "cooow"]) {
+			const textEncoder = new TextEncoder();
+			const keyBuf = textEncoder.encode(value);
+			const key = new ReferencedValue({ offset: 0n, length: 0 }, keyBuf.buffer);
 
-  beforeEach(() => {
-    buffer = new Uint8Array(4096);
-  });
+			const [rv, mp] = await bptree.find(key);
 
-  it("generates an empty tree", async () => {
-    /*
-		let mp: MemoryPointer = {
-			offset: BigInt(0),
-			length: 0,
+			expect(value).toEqual(new TextDecoder().decode(rv.value));
+			expect(mp.offset).toEqual(BigInt(idx));
+			idx += 1;
+			console.log(rv, mp);
+		}
+	});
+
+	it("should find items that were sequentially inserted", async () => {
+		mockRangeResolver = async ({ start, end }) => {
+			const indexFile = await readBinaryFile("bptree_sequential.bin");
+			const slicedPart = indexFile.slice(start, end + 1);
+
+			const arrayBuffer = slicedPart.buffer.slice(
+				slicedPart.byteOffset,
+				slicedPart.byteOffset + slicedPart.byteLength
+			);
+
+			return {
+				data: arrayBuffer,
+				totalLength: arrayBuffer.byteLength,
+			};
 		};
 
-		const tree = new BPTree(resolver, new TestMetaPage(mp), buffer);
+		const page = new testMetaPage({ offset: 8192n, length: 256});
+		const bptree = new BPTree(mockRangeResolver, page, mockDataFileResolver);
+
+		for (let idx = 0; idx < 256; idx++) {
+			const valueBuf = new ArrayBuffer(8);
+			const view = new DataView(valueBuf);
+			view.setBigUint64(0, BigInt(idx));
+
+			const key = new ReferencedValue({ offset: 0n, length: 0 }, valueBuf);
 
-		const res = await tree.find(textEncode("howdy"));
+			const [rv, mp] = await bptree.find(key);
 
-		expect(res).toEqual([{ offset: BigInt(0), length: 0 }, false]);
-		*/
-  });
+			if (rv && rv.value) {
+				const rvValue = new DataView(rv.value).getBigUint64(0);
+				console.log("r: ", rvValue);
+			} else {
+				console.log("value undefined: ", idx);
+				break;
+			}
```

Comment:
for this test case, it seeks and finds upto `idx: 85`. This is because the given root node is from values [0 to 84]. My current intuition is that the first node's page has been reached and it's having a hard time jumping to the next page.

Reply:
Yep that's correct, each page stores ~100 values and if you're storing ints like in the test the first page will split and end at ~85.

Reply:
To jump to the next page you'll have to implement the `.Next()` iterator function to do the traversal.

Reply:
> To jump to the next page you'll have to implement the `.Next()` iterator function to do the traversal.

Do you mean something like: https://github.com/kevmo314/appendable/pull/112/files#diff-0efa68a1779c18545bc009ae1e22699bd9df18732220296ea2e2a6073b3b5d11R45-R101

If this is the right approach, I'll start debugging why it's not behaving accordingly

Reply:
Ah yeah that's right, but in this test you're not calling that iterator in this test. Here's how to call it: https://github.com/kevmo314/appendable/blob/main/pkg/btree/bptree_test.go#L341-L362

Reply:
I thought the `find()` would internally call the iterator: https://github.com/kevmo314/appendable/pull/112/files#diff-2ebc6cac248e11e45734206a7f4f04550ca2b42269743a8a5143b11320eb9072R98-R115

But I may be incorrect.